enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+  - **Mitigation**: We will reject private cluster installations using public IngressControllerLB subnets (see
+    [Consistent Cluster Scope with IngressControllerLB Subnets Validation](#consistent-cluster-scope-with-ingresscontrollerlb-subnets-validation))
+  - **Mitigation**: For private cluster installations using subnet discovery for IngressControllers, public subnets will be excluded from the 
+    `default` IngressControllers, provided the subnets are correctly tagged.
+- **Risk**: Using public ControlPlaneInternalLB subnets which would expose the internal control plane API load balancer.
+  - **Mitigation**: We will reject using public ControlPlaneInternalLB subnets (see
+    [Consistent Cluster Scope with ControlPlaneLB Subnets Validation](#consistent-cluster-scope-with-controlplanelb-subnets-validation)).
+
+### Drawbacks
+
+- Deprecating `platform.aws.subnets` will be painful for users.
+- Distinguishing between BYO subnets/VPC and installer-created subnets/VPC may be confusing if we need
+  to expand the `platform.aws.vpc.subnets` field in the future to accommodate installer-created
+  subnets configuration.
+  - For example, there may be a future desire to specify the IngressController role for installer-created subnets,
+    and it is unclear how this API design would accommodate that.
+- The all-or-nothing approach in [Automatic vs. Manual Role Selection](#automatic-vs-manual-role-selection) prevents
+  users from specifying only a subset of roles to be automatically determined.
+  - For example, a cluster admin might want to isolate the ClusterNode subnet from the ControlPlaneInternalLB subnet,
+    but leave the IngressControllerLB subnets unspecified to rely on subnet discovery for the `default`
+    IngressController. This combination is not currently supported.
+  - While it may be possible to support mixed automatic and manual role selection in the future, the semantics of
+    that configuration remain unclear (see
+    [Allowing Mixed Automatic and Manual Subnet Selection Alternative](#allowing-mixed-automatic-and-manual-subnet-selection-alternative)).
+
+## Open Questions
+
+- Q: Should the install-config design assume all IngressControllerLB subnets are ClusterNode subnets too?
+  - A: No, we must support dedicated IngressController subnet use cases.
+- Q: When using manual role selection, what subnet does the bootstrap node use? Do we need another role?
+- Q: Should we require the `OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY` flag to be set when using `platform.aws.vpc.subnets`
+  to assign ClusterNode roles to public subnets? Or should we disallow public ClusterNode subnets, and only enable
+  `OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY`functionality in automatic role selection mode? Or a combination of both?
+
+## Test Plan
+
+### Installer Testing
+
+E2E test(s) will also be added to the installer to verify functionality of the new [Installer API](#installer-updates).
```

Comment:
I think we should introduce a specific test that separates LBs onto a separate subnet from cluster nodes.